classes/Cart.php

```diff
@@ -970,6 +970,10 @@ protected function applyProductCalculations($row, $shopContext, $productQuantity
                 break;
         }
 
+        // Update unit price in case cart reductions happened
+        $row['unit_price'] = $row['unit_price_tax_excluded'] = $row['unit_price_ratio'] != 0 ? $row['price_with_reduction_without_tax'] / $row['unit_price_ratio'] : 0.0;
+        $row['unit_price_tax_included'] = $row['unit_price_ratio'] != 0 ? $row['price_with_reduction'] / $row['unit_price_ratio'] : 0.0;
```

Comment:
@jolelievre could you test it with the gift automatically added to the cart that has the unit price?